andela/ah-kgl-avengers-frontend
# #165020167 Create, read, and update article(CRU)

### What does this PR do?

- Adds CREATE, READ and UPDATE operations on article

### Description of Task to be completed?

- user should be able to create an article
- user should be able to read an article
- user should be able to update an article
- user should be able to delete an article

### How should this be manually tested?

- Clone the repository locally by running ` git clone https://github.com/andela/ah-kgl-avengers-frontend.git`
- Check out to this branch by running `git checkout ft-article-crud-165020167`
- Install dependencies by running `npm install`
- Update the *.env* file according to the *.env.sample* file
- Add a valid token in the *initialState.js* file
- Start the server by running `npm start``
- Browse to `localhost:8000/new-post` to get the article editor
- Create an article and publish it finally by `clicking on the Publish button`
- You will be redirected to `localhost:8000/articles/:slug` to read your published article

### Any background context you want to provide?

- User has to be logged in to create an article and publish it but reading an article doesn't require a token

- Deleting an article is not yet implemented as we need to add **dashboard** for that

### What are the relevant pivotal tracker stories?

[Users should be able to create, read, update and delete articles #165020167](https://www.pivotaltracker.com/n/projects/2321840/stories/165020167)

### Screenshots (if appropriate)
- **Read article**
<img width="1358" alt="Screen Shot 2019-06-14 at 00 35 41" src="https://user-images.githubusercontent.com/24508902/59471974-2c5dba00-8e3d-11e9-8244-e6f8b2e4d37b.png">
<img width="1324" alt="Screen Shot 2019-06-14 at 00 35 51" src="https://user-images.githubusercontent.com/24508902/59471975-2cf65080-8e3d-11e9-801f-8c7f226dee1c.png">

- **Create an article**
<img width="1388" alt="Screen Shot 2019-06-14 at 00 35 16" src="https://user-images.githubusercontent.com/24508902/59472003-3ed7f380-8e3d-11e9-83f9-9f8d48638f37.png">


### Questions:
none
